contracts/carrot-app/src/handlers/execute.rs

```diff
@@ -51,38 +50,24 @@ fn create_position(
     env: Env,
     info: MessageInfo,
     app: App,
-    mut create_position_msg: CreatePositionMessage,
+    create_position_msg: CreatePositionMessage,
 ) -> AppResult {
     // TODO verify authz permissions before creating the position
     app.admin.assert_admin(deps.as_ref(), &info.sender)?;
-    let mut response = app.response("create_position");
     // We start by checking if there is already a position
-    let funds = create_position_msg.funds;
-    let funds_to_deposit = if get_osmosis_position(deps.as_ref()).is_ok() {
-        let (withdraw_msg, withdraw_amount, total_amount, withdrawn_funds) =
-            _inner_withdraw(deps.as_ref(), &env, None, &app)?;
-
-        response = response
-            .add_message(withdraw_msg)
-            .add_attribute("withdraw_amount", withdraw_amount)
-            .add_attribute("total_amount", total_amount);
-
-        // We add the withdrawn funds to the input funds
-        let mut coins: Coins = funds.try_into()?;
-        for fund in withdrawn_funds {
-            coins.add(fund)?;
+    if let Ok(position) = get_osmosis_position(deps.as_ref()) {
+        // We only care if the position has liquidity
+        if position.position.unwrap().liquidity != "0" {
+            return Err(AppError::PositionExists {});
         }
```

Comment:
```suggestion
    if get_osmosis_position(deps.as_ref()).is_ok() {
        return Err(AppError::PositionExists {});
```
Pretty sure position can't exist if liquidity is zero

Reply:
It can if people people withdraw all using the raw position messages no ?

Reply:
> It can if people people withdraw all using the raw position messages no ?

Expanded a manual withdrawal test, to ensure position deleted from the pool